tests/test_spectral_profile.py

```diff
@@ -0,0 +1,303 @@
+        )
+        assert np.isclose(
+            normalization_factor_spin_polarized, 0.5 * normalization_factor
+        )
+
+    def test_extract_band_gap(self):
+        """
+        Test the `extract_band_gap` method.
+        """
+        # ! add test when `ElectronicEigenvalues` is implemented
+        pass
+
+    def test_resolve_pdos_name(self, simulation_electronic_dos: Simulation):
+        """
+        Test the `resolve_pdos_name` method.
+        """
+        # Get projected DOSProfile from the simulation fixture
+        projected_dos = (
+            simulation_electronic_dos.outputs[0].electronic_dos[0].projected_dos
+        )
+        assert len(projected_dos) == 3
+        pdos_names = ['orbital s Ga', 'orbital px As', 'orbital py As']
+        for i, pdos in enumerate(projected_dos):
+            name = pdos.resolve_pdos_name(logger)
+            assert name == pdos_names[i]
+
+    def test_extract_projected_dos(self, simulation_electronic_dos: Simulation):
+        """
+        Test the `extract_projected_dos` method.
+        """
+        # Get Outputs and ElectronicDensityOfStates from the simulation fixture
+        outputs = simulation_electronic_dos.outputs[0]
+        electronic_dos = outputs.electronic_dos[0]
+
+        # Initial tests for the passed `projected_dos` (only orbital PDOS)
+        assert len(electronic_dos.projected_dos) == 3  # only orbital projected DOS
+        orbital_projected = electronic_dos.extract_projected_dos('orbital', logger)
+        atom_projected = electronic_dos.extract_projected_dos('atom', logger)
+        assert len(orbital_projected) == 3 and len(atom_projected) == 0
+        orbital_projected_names = [orb_pdos.name for orb_pdos in orbital_projected]
+        assert orbital_projected_names == [
+            'orbital s Ga',
+            'orbital px As',
+            'orbital py As',
+        ]
+        assert (
+            orbital_projected[0].entity_ref
+            == outputs.model_system_ref.cell[0].atoms_state[0].orbitals_state[0]
+        )  # orbital `s` in `Ga` atom
+        assert (
+            orbital_projected[1].entity_ref
+            == outputs.model_system_ref.cell[0].atoms_state[1].orbitals_state[0]
+        )  # orbital `px` in `As` atom
+        assert (
+            orbital_projected[2].entity_ref
+            == outputs.model_system_ref.cell[0].atoms_state[1].orbitals_state[1]
+        )  # orbital `py` in `As` atom
+
+        orbital_projected = electronic_dos.extract_projected_dos('orbital', logger)
+        atom_projected = electronic_dos.extract_projected_dos('atom', logger)
+        assert len(orbital_projected) == 3 and len(atom_projected) == 0
+
+    @pytest.mark.parametrize(
+        'value, result',
+        [
+            (None, [1.5, 1.2, 0, 0, 0, 0.8, 1.3]),
+            ([1.5, 1.2, 0, 0, 0, 0.8, 1.3], [1.5, 1.2, 0, 0, 0, 0.8, 1.3]),
+            ([30.5, 1.2, 0, 0, 0, 0.8, 1.3], [30.5, 1.2, 0, 0, 0, 0.8, 1.3]),
+        ],
```

Comment:
What is the aim of the middle test? The first / last one checks for a DOS built up from pDOS / fully determined by itself.

Reply:
Indeed, it is pretty dumb. I deleted it.